DOCSUP-11556: Document the Pad functions (#27304)
ClickHouse/ClickHouse: docs/en/sql-reference/functions/string-functions.md

Returns the length of a string in Unicode code points (not in characters), assuming that the string contains a set of bytes that make up UTF-8 encoded text. If this assumption is not met, it returns some result (it does not throw an exception).
The result type is UInt64.

## leftPad {#leftpad}

Pads the current string from the left with spaces or a specified string (multiple times, if needed) until the resulting string reaches the given length. Similarly to the MySQL `LPAD` function.

**Syntax**

``` sql
leftPad('string', 'length'[, 'pad_string'])
```

**Arguments**

> -   `string` — Input string, that needs to be padded. [String](../data-types/string.md).

Reviewer: ```suggestion
-   `string` — Input string that needs to be padded. [String](../data-types/string.md).
```
A comma before 'that' is usually not needed.

-   `length` — The length of the resulting string. [UInt](../data-types/int-uint.md). If the value is less than the input string length, then the input string is returned as-is.
-   `pad_string` — The string to pad the input string with. [String](../data-types/string.md). Optional. If not specified, then the input string is padded with spaces.

**Returned value**

-   The resulting string of the given length.

Type: [String](../data-types/string.md).

**Example**

Query:

``` sql
SELECT leftPad('abc', 7, '*'), leftPad('def', 7);
```

Result:

``` text
┌─leftPad('abc', 7, '*')─┬─leftPad('def', 7)─┐
│ ****abc                │     def           │
└────────────────────────┴───────────────────┘
```

## leftPadUTF8 {#leftpadutf8}

Pads the current string from the left with spaces or a specified string (multiple times, if needed) until the resulting string reaches the given length. Similarly to the MySQL `LPAD` function. While in the [leftPad](#leftpad) function the length is measured in bytes, here in the `leftPadUTF8` function it is measured in code points.

**Syntax**

``` sql
leftPadUTF8('string','length'[, 'pad_string'])
```

**Arguments**

> -   `string` — Input string, that needs to be padded. [String](../data-types/string.md).

Reviewer: ```suggestion
-   `string` — Input string that needs to be padded. [String](../data-types/string.md).
```
A comma before 'that' is usually not needed.

-   `length` — The length of the resulting string. [UInt](../data-types/int-uint.md). If the value is less than the input string length, then the input string is returned as-is.
-   `pad_string` — The string to pad the input string with. [String](../data-types/string.md). Optional. If not specified, then the input string is padded with spaces.

**Returned value**

-   The resulting string of the given length.

Type: [String](../data-types/string.md).

**Example**

Query:

``` sql
SELECT leftPadUTF8('абвг', 7, '*'), leftPadUTF8('дежз', 7);
```

Result:

``` text
┌─leftPadUTF8('абвг', 7, '*')─┬─leftPadUTF8('дежз', 7)─┐
│ ***абвг                     │    дежз                │
└─────────────────────────────┴────────────────────────┘
```

## rightPad {#rightpad}

Pads the current string from the right with spaces or a specified string (multiple times, if needed) until the resulting string reaches the given length. Similarly to the MySQL `RPAD` function.

**Syntax**

``` sql
rightPad('string', 'length'[, 'pad_string'])
```

**Arguments**

> -   `string` — Input string, that needs to be padded. [String](../data-types/string.md).

Reviewer: ```suggestion
-   `string` — Input string that needs to be padded. [String](../data-types/string.md).
```
A comma before 'that' is usually not needed.

-   `length` — The length of the resulting string. [UInt](../data-types/int-uint.md). If the value is less than the input string length, then the input string is returned as-is.
-   `pad_string` — The string to pad the input string with. [String](../data-types/string.md). Optional. If not specified, then the input string is padded with spaces.

**Returned value**

-   The resulting string of the given length.

Type: [String](../data-types/string.md).

**Example**

Query:

``` sql
SELECT rightPad('abc', 7, '*'), rightPad('abc', 7);
```

Result:

``` text
┌─rightPad('abc', 7, '*')─┬─rightPad('abc', 7)─┐
│ abc****                 │ abc                │
└─────────────────────────┴────────────────────┘
```

## rightPadUTF8 {#rightpadutf8}

Pads the current string from the right with spaces or a specified string (multiple times, if needed) until the resulting string reaches the given length. Similarly to the MySQL `RPAD` function. While in the [rightPad](#rightpad) function the length is measured in bytes, here in the `rightPadUTF8` function it is measured in code points.

**Syntax**

``` sql
rightPadUTF8('string','length'[, 'pad_string'])
```

**Arguments**

> -   `string` — Input string, that needs to be padded. [String](../data-types/string.md).

Reviewer: ```suggestion
-   `string` — Input string that needs to be padded. [String](../data-types/string.md).
```
A comma before 'that' is usually not needed.

Returns the length of a string in Unicode code points (not in characters), assuming that the string contains a set of bytes that make up UTF-8 encoded text. If this assumption is not met, it returns some result (it does not throw an exception).
The result type is UInt64.

## leftPad {#leftpad}

Pads the current string from the left with a specified string (multiple times, if needed) until the resulting string reaches the given length. Similarly to the MySQL `LPAD` function.

**Syntax**

``` sql

> leftPad('string','length', 'pad_string')

Reviewer 2: ```suggestion
leftPad('string', 'length', 'pad_string')
```

```

**Arguments**

> -   `string` — Input string, that need to be padded. [String](../data-types/string.md).

Reviewer 2: ```suggestion
-   `string` — Input string, that needs to be padded. [String](../data-types/string.md).
```

> -   `length` — The length of the resulting string once the input string pads. [UInt](../data-types/int-uint.md). If the value is less than input string length, then string is returned as-is.

Reviewer 2: ```suggestion
-   `length` — The length of the resulting string. [UInt](../data-types/int-uint.md). If the value is less than the input string length, then the input string is returned as-is.
```

> -   `pad_string` — The string to pad the current input string with. [String](../data-types/string.md).

Reviewer 2: ```suggestion
-   `pad_string` — The string to pad the input string with. [String](../data-types/string.md). Optional. If not specified, then the input string is padded with spaces.
```

> [String](../data-types/string.md)

Reviewer 2: ```suggestion

```

**Returned value(s)**

> -   The resulting string reaches the given length.

Reviewer 2: ```suggestion
-   The resulting string with the given length.
```

Type: [String](../data-types/string.md).

**Example**

Query:

``` sql

> SELECT leftPad('abc', 7, '*');

Reviewer 2: ```suggestion
SELECT leftPad('abc', 7, '*'), leftPad('def', 7);
```

```

Result:

``` text

> ┌─leftPad('abc', 7, '*')─┐
> │ ****abc                │
> └────────────────────────┘

Reviewer 2: ```suggestion
┌─leftPad('abc', 7, '*')─┬─leftPad('def', 7)─┐
│ ****abc                │     def           │
└────────────────────────┴───────────────────┘
```

```

## leftPadUTF8 {#leftpadutf8}

> Pads the current string from the left with a specified string (multiple times, if needed) until the resulting string reaches the given length. Similarly to the MySQL `LPAD` function.

Reviewer 2: ```suggestion
Pads the current string from the left with a specified string (multiple times, if needed) until the resulting string reaches the given length. Similarly to the MySQL `LPAD` function. While in the [leftPad](#leftpad) function the length is measured in bytes, here in the `leftPadUTF8` function it is measured in code points.
```

**Syntax**

``` sql

> leftPadUTF8('string','length', 'pad_string')

Reviewer 2: ```suggestion
leftPadUTF8('string','length'[, 'pad_string'])
```

```

**Arguments**

> -   `string` — Input UTF-8 string, that need to be padded. [String](../data-types/string.md).

Reviewer 2: ```suggestion
-   `string` — Input UTF-8 string, that needs to be padded. [String](../data-types/string.md).
```

> -   `length` — The length of the resulting string once the input string pads. [UInt](../data-types/int-uint.md). If the value is less than input string length, then string is returned as-is.

Reviewer 2: ```suggestion
-   `length` — The length of the resulting string. [UInt](../data-types/int-uint.md). If the value is less than the input string length, then the input string is returned as-is.
```

> -   `pad_string` — The string to pad the current input string with. [String](../data-types/string.md).

Reviewer 2: ```suggestion
-   `pad_string` — The string to pad the current input string with. [String](../data-types/string.md). Optional. If not specified, then the input string is padded with spaces.
```

> [String](../data-types/string.md)

Reviewer 2: ```suggestion

```

**Returned value(s)**

> -   The resulting UTF-8 string reaches the given length.

Reviewer 2: ```suggestion
-   The resulting UTF-8 string with the given length.
```

Type: [String](../data-types/string.md).

**Example**

Query:

``` sql

> SELECT leftPadUTF8('абвг', 7, '*');

Reviewer 2: ```suggestion
SELECT leftPadUTF8('абвг', 7, '*'), leftPadUTF8('дежз', 7);
```

```

Result:

``` text

> ┌─leftPadUTF8('абвг', 7, '*')─┐
> │ ***абвг                     │
> └─────────────────────────────┘

Reviewer 2: ```suggestion
┌─leftPadUTF8('абвг', 7, '*')─┬─leftPadUTF8('дежз', 7)─┐
│ ***абвг                     │    дежз                │
└─────────────────────────────┴────────────────────────┘
```
